vernac/obligations.ml

```diff
@@ -20,6 +20,12 @@ open Pp
 open CErrors
 open Util
 
+(* EJGA: This should go away, no more need for fix_exn  *)
+module Stm = struct
+    let u = (fun x -> x)
+    let get_fix_exn () = u
+end
+
```

Comment:
If I get it right, by upstream you mean "fixing the code of obligations".
This is what is done, in a dirty way.  The piece of data obligations needs to know, in order to attach to the future the right state id, is the one available in the ref.  AFAICT obligations is the only piece of code with such need, so we did not bother coming up with a saner general schema.